src/sssom/util.py

```diff
@@ -246,7 +246,7 @@ def clean_prefix_map(self, strict: bool = True) -> None:
         if self.metadata:
             prefixes_in_table.update(get_prefixes_used_in_metadata(self.metadata))
 
-        missing_prefixes = prefixes_in_table - self.converter.get_prefixes()
+        missing_prefixes = prefixes_in_table - self.converter.get_prefixes(include_synonyms=True)
```

Comment:
Maybe i am wrong. I added a test below that illustrates the issue. 

Lets say the raw data (prefixes in table) contains `snomed`. The converter has `SCTID` as a prefix. `snomed` as a synonym. Now this line of code here will not include `snomed` in missing prefixes, and move on. 

Now you go a few lines below to extract a subconverter over all the prefixes in the table, including `snomed`, which will now become the main converter (this part is not very relevant for the problem but bear with me). Now, when the prefixmap gets serialsed to a valid SSSOM file, which uses `msdf.converter.bimap` (see write_table() method). This map does _not_ contain `snomed`. so it will write SCTID, while the data (the mappings) still contains `snomed`. 

If you still think this PR is right, you need to convince me that giving a mapping set with a value containing `snomed`, and `snomed` being a synonym in the EPM, the serialised SSSOM files contains `snomed` prefix..

Reply:
Mm, I see the problem. I think you touched on this at the last meeting. How do you feel about serializing a non-bijective prefix map which would include the synonyms, then?

When it comes to working with synonyms in SSSOM, do you think this we've covered all of the cases? Or are there likely to be more places where synonym coverage is needed?